player/lua/osc.lua

```diff
@@ -651,47 +651,44 @@ end
 -- Message display
 --
 
+-- pos is 1 based
 function limited_list(prop, pos)
-    local fs = tonumber(mp.get_property('options/osd-font-size'))
-    local max_items = math.ceil(720 / fs)
-
     local proplist = mp.get_property_native(prop, {})
     local count = #proplist
-    if (count == 0) then
+    if count == 0 then
         return count, proplist
     end
-    local min, max = 0, count
-    local temp = {}
-    if (count > max_items - 1) then
-        local extra = math.ceil(max_items / 2) - 1
-        min = pos - extra
-        max = pos + extra
-        if (pos < extra) then
-            max = max + math.abs(min)
-            min = 0
-        end
+
+    local fs = tonumber(mp.get_property('options/osd-font-size'))
+    local max = math.ceil(720 / fs)
+    if max % 2 == 0 then
+        max = max - 1
     end
+    local delta = math.ceil(max / 2) - 1
+    local begi = math.max(math.min(pos - delta, count - max + 1), 1)
+    local endi = math.min(begi + max - 1, count)
 
-    for i=min+1, math.min(max, count) do
+    local reslist = {}
+    for i=begi, endi do
         local item = proplist[i]
-        item.current = (i-1 == pos) and true or nil
-        table.insert(temp, item)
+        item.current = (i == pos) and true or nil
+        table.insert(reslist, item)
     end
-    return count, temp
+    return count, reslist
 end
 
 function get_playlist()
-    local pos = mp.get_property_number('playlist-pos')
+    local pos = mp.get_property_number('playlist-pos') + 1
```

Comment:
Yes, I know ... and had it there before. But for consistency with chapter I've changed it back :|.